### PR TITLE
Jpi variant rule/support ivy module resolve metadata

### PR DESCRIPTION
**Pull request by rpalcolea**
Hi @sghill, 

This fixes https://github.com/jenkinsci/gradle-jpi-plugin/issues/142

### CODE REVIEW

**Comment by rpalcolea** [comment]
Interesting, I can't replicate the codenarc failure locally

```

 ~/Projects/github/jenkinsci/gradle-jpi-plugin  on JpiVariantRule/support-IvyModuleResolveMetadata !1 ▓▒░·················░▒▓ 10752  at 16:03:30  1.8.0_181-b02 ☕ 
❯ ./gradlew clean codenarcMain

BUILD SUCCESSFUL in 4s
2 actionable tasks: 2 executed
```

Is it possible to re trigger the jenkins job?

**Comment by rpalcolea** [comment]
Interesting, I can't replicate the codenarc failure locally

```

 ~/Projects/github/jenkinsci/gradle-jpi-plugin  on JpiVariantRule/support-IvyModuleResolveMetadata !1 ▓▒░·················░▒▓ 10752  at 16:03:30  1.8.0_181-b02 ☕ 
❯ ./gradlew clean codenarcMain

BUILD SUCCESSFUL in 4s
2 actionable tasks: 2 executed
```

Is it possible to re trigger the jenkins job?

File: src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiVariantRule.groovy [hunk 1]
```diff
@@ -17,7 +18,9 @@ abstract class JpiVariantRule implements ComponentMetadataRule {
     @Override
     void execute(ComponentMetadataContext ctx) {
         def id = ctx.details.id
-        if (isJenkinsPackaging(ctx)) {
+        if(isIvyResolvedDependency(ctx)) {
```
Comment (sghill) [inline src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiVariantRule.groovy]: This is failing codenarc with:

> The if keyword within class org.jenkinsci.gradle.plugins.jpi.JpiVariantRule is not followed by a single space

File: src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiVariantRule.groovy [hunk 2]
```diff
@@ -65,6 +68,10 @@ abstract class JpiVariantRule implements ComponentMetadataRule {
         // This variant might still resolve to a jar file - https://github.com/gradle/gradle/issues/11974
     }
 
+    private boolean isIvyResolvedDependency(ComponentMetadataContext ctx) {
+        return ctx.metadata instanceof IvyModuleResolveMetadata
```
Comment (sghill) [inline src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiVariantRule.groovy]: This is failing codenarc with:

> The return keyword is not needed and can be removed

File: src/test/groovy/org/jenkinsci/gradle/plugins/jpi/JpiIntegrationSpec.groovy [hunk 3]
```diff
@@ -330,4 +330,64 @@ class JpiIntegrationSpec extends IntegrationSpec {
         new File(projectDir.root, "build/testRepo/org/jenkinsci/sample/${projectName}/1.0/${projectName}-1.0-javadoc.jar").exists()
         new File(projectDir.root, "build/testRepo/org/jenkinsci/sample/${projectName}/1.0/${projectName}-1.0-sources.jar").exists()
     }
+
+    def 'handles dependencies coming from ivy repository'() {
```
Comment (sghill) [inline src/test/groovy/org/jenkinsci/gradle/plugins/jpi/JpiIntegrationSpec.groovy]: :sparkles: Thanks for this!

File: src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiVariantRule.groovy (hunk 2 above)
Comment (rpalcolea) [reply src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiVariantRule.groovy]: I’ll fix it tonight

**Comment by rpalcolea** [comment]
@sghill.

https://github.com/jenkinsci/gradle-jpi-plugin/pull/144/commits/fc296c1b966d5e3e2f0b9a44d7f4a29b47e88313 fixes the codenarc failures

**Comment by rpalcolea** [comment]
Thanks @jjohannes! I'll keep an eye on that issue

**Comment by rpalcolea** [comment]
Hi @sghill , thanks for merging this! are you planning to do a release soon? :)

**Pull request merged**